inventory_management_system_api/core/database.py

```diff
@@ -28,4 +32,27 @@ def get_database() -> Database:
     return mongodb_client[db_config.name.get_secret_value()]
 
 
+@contextmanager
+def start_session_transaction(action_description: str) -> Generator[ClientSession, None, None]:
+    """
+    Starts a MongoDB session followed by a transaction and returns the session to use.
+
+    Also handles write conflicts.
+
+    :param action_description: Description of what the transaction is doing so it can be used in any raised errors.
+    :raises WriteConflictError: If there a write conflict during the transaction.
+    """
```

Comment:
```suggestion
def start_session_transaction(action_description: str) -> Generator[ClientSession, None, None]:
    """
    Starts a MongoDB session followed by a transaction and returns the session to use.

    Also handles write conflicts.

    :param action_description: Description of what the transaction is doing so it can be used in any raised errors.
    :raises WriteConflictError: If there a write conflict during the transaction.
    :yields/returns?: A MongoDB session
    """
```

Reply:
Yeah, not sure my self. https://stackoverflow.com/questions/7652540/docstring-tag-for-yield-keyword suggests return so will go with that.